### DISCUSSION TITLE
Moon's apparent angular diameter

### DISCUSSION THREAD

**Discussion by offsky** [cosinekitty/astronomy#135] [Q&A] [answered]
First off, thanks for adding the Moon libration and apparent diameter to your excellent library.  This will be very useful!  

I was hoping that you could help me understand the calculation you are doing for the apparent diameter.  Previously, I was doing this calculation myself and I was doing `atan(moon_radius / earth_moon_distance)`.  This matches what nasa calculates (https://svs.gsfc.nasa.gov/4874) and the trigonometry is as simple as you can get.  Then I looked at your new code and I see you are doing something different.

`Math.atan(MOON_MEAN_RADIUS_KM / Math.sqrt(dist_km * dist_km - MOON_MEAN_RADIUS_KM * MOON_MEAN_RADIUS_KM));`

I looks like you are doing a Pythagorean theorem (with dist_km being the hypotenuse of the triangle) to get a slightly different dist_km. In the grand picture, it actually makes no difference as my calculation and your calculation are the same to 3 decimal places, but for my own education I would love to know why are you doing it the way you are.

Question 2: Since the observer will always be on the surface of earth, why is the earths radius not subtracted from the moon-earth distance to get a more accurate calculation of the apparent observed angle?  By my math, this makes a difference of up to 1 arcmin, but this will no longer match NASAs calculations. Is this just convention or too small to matter.

**Comment by cosinekitty** [comment, marked as answer]
Those are great questions! For Question 1, it really doesn't matter because, like you say, you get about the same answer either way. The way I do it is a little more accurate for objects large enough to be resolved by the naked eye. Here is a diagram for illustration purposes:

![apparent_angular_diameter](https://user-images.githubusercontent.com/11699954/140820305-c7312c9e-5211-4938-aea4-379aa0bae208.png)

Here, *a* represents half of the apparent angular diameter of the spherical body on the right, and marks the location of the observer. *D* is the distance between the observer and the center of the observed body. *R* is the radius of the body. The the formula `a = atan(R/D)` is a good approximation, but it breaks down as *D* gets smaller. In the extreme, imagine you are in a lunar lander approaching the surface of the moon. When you land, you know the apparent angular diameter of the moon should be 180 degrees. At that point, *R* = *D*, and the simple formula gives you an incorrect answer of atan(1) = 45 degrees for the apparent angular radius, or 90 degrees for the angular diameter.

The correction is to use the Pythagorean theorem to find *x*, which is the actual tangent line of sight to the visible edge of the object. (Consider that you never quite see half of any spherical object. You always see less than half, especially as you get closer to it.) *R* and *x* are legs of a right triangle, and *D* is the hypotenuse. Then `a = atan(R/x)` always gives the correct angular radius, even when you land on the moon.

About Question 2, you are absolutely right that the geocentric observer is not realistic. But there is no single correct answer that works for all observers on the surface of the Earth. People at different locations will all be different distances from the Moon, giving slightly different angular sizes. It doesn't make much difference, but if such accuracy were needed for some reason, there is a function [ObserverVector](https://github.com/cosinekitty/astronomy/tree/master/source/js#ObserverVector) you can use to calculate a vector from the center of the Earth to any location on the surface of the Earth. Then you could calculate the exact distance to the Moon and refine the answer.

I hope this helps it all make sense.

**Comment by offsky** [comment]
Ahh, that makes perfect sense now. Thanks for teaching me something new!  

Regarding question 2, I suppose if you really wanted to be perfect you would take the observers location into account and also the refraction of the atmosphere for when the moon is near the horizon and you would have to calculate an apparent diameter in both the x and y dimensions. Overkill, but good to know it can be done if I ever wanted to :)
